textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/nlpextrator/NlpExtractor.java

```diff
@@ -93,64 +132,114 @@ public ITuple getNextTuple() throws Exception {
+     * If the flag is "NE_ALL", we set up the NamedEntityTagAnnotator,
+     * if it's "POS", then only PartOfSpeechAnnotator is needed.
+     * <p>
+     * The pipeline has to be this order: TokenizerAnnotator,
      * SentencesAnnotator, PartOfSpeechAnnotator, LemmaAnnotator and
-     * NamedEntityTagAnnotator. The order is mandatory because they have
-     * dependencies. After the pipeline, each token is wrapped as a CoreLabel
+     * NamedEntityTagAnnotator.
+     * <p>
+     * In the pipeline, each token is wrapped as a CoreLabel
      * and each sentence is wrapped as CoreMap. Each annotator adds its
      * annotation to the CoreMap(sentence) or CoreLabel(token) object.
      * <p>
-     * After the pipeline is done, scan each CoreLabel(token) for
-     * its NamedEntityAnnotation, if it's a valid value (not 'O'),
-     * then makes it a span and add to the return list.
-     * The Stanford NLP constants are mapped into the NE constants.
+     * After the pipeline, scan each CoreLabel(token) for its
+     * NamedEntityAnnotation or PartOfSpeechAnnotator depends on the flag
+     * <p>
+     * For each Stanford NLP annotation, get it's corresponding inputNlpTokenType
+     * that used in this package, then check if it equals to the input constant.
+     * If yes, makes it a span and add to the return list.
+     * <p>
      * The NLP package has annotations for the start and end position of a token
      * and it perfectly matches the span design so we just use them.
-     *
+     * <p>
+     * For Example: With TextField value: "Microsoft, Google and Facebook are
+     * organizations while Donald Trump and Barack Obama are persons", with
+     * fieldName: Sentence1 and inputConstant is Organization. Since the
+     * inputConstant require us to use NamedEntity Annotator in the Stanford
+     * NLP package, the flag would be set to "NE".
+     * The pipeline would set up to cover the Named Entity Recognizer. Then
+     * get the value of NamedEntityTagAnnotation for each CoreLabel(token).If
+     * the value is the constant "Organization", then it meets the
+     * requirement. In this case "Microsoft","Google" and "Facebook" will
+     * satisfy the requirement. "Donald Trump" and "Barack Obama" would
+     * have constant "Person" and do not meet the requirement. For each
+     * qualified token, create a span accordingly and add it to the returned
+     * list. In this case, token "Microsoft" would be span:
+     * ["Sentence1", 0, 9, Organization, "Microsoft"]
      */
-    private List<Span> extractNESpans(IField iField, String fieldName) {
+    private List<Span> extractNlpSpans(IField iField, String fieldName) {
         List<Span> spanList = new ArrayList<>();
         String text = (String) iField.getValue();
         Properties props = new Properties();
-        props.setProperty("annotators", "tokenize, ssplit, pos, lemma, ner");
+
+        //Setup Stanford NLP pipelien based on flag
```

Comment:
"pipelien" -> "pipeline". @sam0227 I think you can avoid this type of mistakes :-)